### PR TITLE
Fix FCS problem with duplicate  ByteFile

### PR DESCRIPTION
**Pull request by dsyme**
We had a report that too many duplicate ByteFile objects are still being created when FCS is used (e.g. in Ionide and Rider), despite #4586

This is because the default FCS configuration for reading memory is to use a ByteFile for the metadata section of metadata-only binaries.  This can be a WeakByteFile to reduce memory usage.

Also the strongly-held cache was set to zero - now that binary readers are more memory efficient and release resources by default we can increase this to a non-zero default.

The screen shot that convinced me this is a problem is from @AviAvni below

image

### CODE REVIEW

File: src/absil/ilread.fs [hunk 1]
```diff
@@ -37,7 +37,7 @@ let _ = if checking then dprintn "warning : ILBinaryReader.checking is on"
 let noStableFileHeuristic = try (System.Environment.GetEnvironmentVariable("FSharp_NoStableFileHeuristic") <> null) with _ -> false
 let alwaysMemoryMapFSC = try (System.Environment.GetEnvironmentVariable("FSharp_AlwaysMemoryMapCommandLineCompiler") <> null) with _ -> false
 let stronglyHeldReaderCacheSizeDefault = 30
-let stronglyHeldReaderCacheSize = try match System.Environment.GetEnvironmentVariable("FSharp_StronglyHeldBinaryReaderCacheSize") with null -> stronglyHeldReaderCacheSizeDefault | s -> int32 s with _ -> stronglyHeldReaderCacheSizeDefault
+let stronglyHeldReaderCacheSize = try (match System.Environment.GetEnvironmentVariable("FSharp_StronglyHeldBinaryReaderCacheSize") with null -> stronglyHeldReaderCacheSizeDefault | s -> int32 s) with _ -> stronglyHeldReaderCacheSizeDefault
```
Comment (forki) [inline src/absil/ilread.fs]: can you please put it on multiple lines? I think this is super hard to follow

**Comment by forki** [comment]
yeah the clippy in the screenshot would have convinced me as well :-)

**Comment by dsyme** [comment]
> yeah the clippy in the screenshot would have convinced me as well :-)

OMG. @AviAvni Please explain :)

**Comment by AviAvni** [comment]
@dsyme @forki clippy help me to solve hard problems ;) 
https://marketplace.visualstudio.com/items?itemName=ErosFratini.ClippyVS
